xarray/tests/test_dataset.py

```diff
@@ -3971,6 +3971,10 @@ def test_ipython_key_completion(self):
             ds.data_vars[item]  # should not raise
         assert sorted(actual) == sorted(expected)
 
+    def test_dir(self):
+        ds = create_test_data(seed=1)
+        assert 'T' not in dir(ds)
```

Comment:
Can you move this to `test_dataset_transpose`, so we're sure to clean it up at the same time?
https://github.com/pydata/xarray/blob/d016ea7d925d3d53796b7c68b6d377bdb7497a8e/xarray/tests/test_dataset.py#L3585-L3586